web/ui/react-app/src/pages/rules/RulePanel.tsx

```diff
@@ -0,0 +1,53 @@
+import React, { FC, CSSProperties, Fragment } from 'react';
+import { Rule } from '../../types/types';
+import { Link } from '@reach/router';
+import { createExpressionLink } from '../../utils';
+
+export const RulePanel: FC<{ rule: Rule; styles?: CSSProperties; tag?: keyof JSX.IntrinsicElements }> = ({
+  rule,
+  styles = {},
+  tag,
+}) => {
+  const Tag = tag || Fragment;
+  const { name, query, labels, annotations, alerts } = rule;
+  let customTagProps = {};
+  const mergedStyles = { background: '#f5f5f5', ...styles };
+  if (tag) {
+    customTagProps = { style: mergedStyles };
+  }
+  return (
+    <Tag {...customTagProps}>
+      <pre className="m-0" style={tag ? undefined : mergedStyles}>
+        <code>
+          <div>
+            {rule.alerts ? 'name: ' : 'record: '}
+            <Link to={createExpressionLink(`ALERTS{alertname="${name}"}`)}>{name}</Link>
+          </div>
+          <div>
+            expr: <Link to={createExpressionLink(query)}>{query}</Link>
+          </div>
+          {alerts && (
```

Comment:
Labels also exist on recording rules, and alerts also need their `for` (duration) field displayed (e.g. `for: 5m`).

I realize now that I overlooked some of this in both the earlier /rules and /alerts PRs, good to catch it now.